llvm/lib/IR/Type.cpp

```diff
@@ -845,6 +852,18 @@ static TargetTypeInfo getTargetTypeInfo(const TargetExtType *Ty) {
     return TargetTypeInfo(ScalableVectorType::get(Type::getInt1Ty(C), 16),
                           TargetExtType::HasZeroInit);
 
+  // RISCV vector tuple type. The layout is represented as the type that needs
+  // the same number of vector registers(VREGS) as this tuple type, represented
+  // as <vscale x (VREGS * 8) x i8>.
+  if (Name == "riscv_vec_tuple") {
+    unsigned TotalNumElts =
+        std::max(cast<ScalableVectorType>(Ty->getTypeParameter(0))
+                     ->getMinNumElements(),
+                 8UL) *
```

Comment:
Does `8UL` work here on targets where `uint64_t` isn't `long` and is instead `long long`?

Reply:
No, it doesn't work, let me change it to ULL, thanks for reminding it!

Reply:
You might need to write `std::max<unit64_t>`

Reply:
Got it.

Reply:
getMinNumElements returns `unsigned` instead of `uint64_t` after 6c1e3e49c10718e1d45a9031198af0ebd64fecb4

Reply:
Got it.